src/components/filters/MeasureValueFilter/DropdownBody.tsx

```diff
@@ -114,15 +117,58 @@ class DropdownBodyWrapped extends React.PureComponent<IDropdownBodyProps, IDropd
                     onToChange={this.handleToChange}
                     onEnterKeyPress={this.onApply}
                     disableAutofocus={disableAutofocus}
+                    separators={separators}
                 />
             );
         }
 
         return null;
     };
 
+    private isApplyButtonDisabledForComparison() {
+        const { value = null } = this.state.value;
+
+        if (value === null) {
+            return true;
+        }
+
+        if (this.props.value === null) {
+            return false;
+        }
+
+        return value === this.props.value.value;
+    }
+
+    private isApplyButtonDisabledForRange() {
+        const { from = null, to = null } = this.state.value;
+
+        if (from === null || to === null) {
+            return true;
+        }
+
+        if (this.props.value === null) {
+            return false;
+        }
+
+        return from === this.props.value.from && to === this.props.value.to;
+    }
+
+    private isApplyButtonDisabledForAll() {
+        return this.props.operator === Operator.ALL;
+    }
+
     private isApplyButtonDisabled = () => {
-        return false;
+        const { operator } = this.state;
+
+        if (Model.isComparisonOperator(operator)) {
```

Comment:
This will need to be rebased after BB-2030 where the model is not providing these functions